packages/sdk/src/Config.ts

```diff
@@ -0,0 +1,64 @@
+import { BaseProvider } from '@ethersproject/providers';
+import {
+  Environment,
+  ConfigDomains,
+  EthereumConfig,
+  SubstrateConfig,
+  Resource,
+} from 'types/config';
+
+export class Config {
+  private devnetConfigUrl = 'https://config.develop.buildwithsygma.com/share/';
+  private testnetConfigUrl = 'https://config-server.test.buildwithsygma.com/share/';
+  private mainnetConfigUrl = '';
+
+  public environment!: ConfigDomains;
+  public provider!: BaseProvider;
+
+  public async init(environment: Environment, provider: BaseProvider): Promise<void> {
```

Comment:
I think if you switch the params you can put environment as optional.

Reply:
Also, probably instead of provider accept chainID | (whatever is ID in substrate; parachainID?)